### DISCUSSION TITLE
Is TanStack Form fully compatible with react compiler?

### DISCUSSION THREAD

**Discussion by rodrigofariow** [TanStack/form#968] [Q&A]
At my company, we are thinking of moving away from react-hook-form simply because it breaks the rules of react on many properties returned by their `useForm` and `useController` hooks. 
They are slowly fixing a few things but a lot of them remain and the main maintainer is not making it a priority for now, AFAIK.

So, before fully committing to this library (awesome library btw 💯 ) , we want to know if all returned properties from `useForm`, `useField`, etc are safely memoizable (follow the rules of react), and thus react compiler compatible.

Thank you for your time 🙏 

CC @crutchcorn @tannerlinsley

**Comment by crutchcorn** [comment]
It should be! We have not yet tested with React Compiler, but if you run into problems, please report them and we'll make sure to fix them

**Comment by Aqib-Rime** [comment]
This is an issue I am facing with react compiler enabled in NextJS 15.

```
export function FieldError() {
  const field = useFieldContext();
  const errors = field.state.meta.errors;

  return (
    <>
      {errors.length > 0 ? (
        <div className="text-red-500 text-xs ml-2">{errors[0].message}</div>
      ) : null} 
    </>
  );
}
```

this code doesn't re-render when compiler is on
